### PR TITLE
Adding Warning for layers outside the current project home directory

### PR DESCRIPTION
**Pull request by SeqLaz**
Adding Warning for layers outside the current project home directory

image

### CODE REVIEW

File: libqfieldsync/project_checker.py [hunk 1]
```diff
@@ -410,3 +415,22 @@ def check_layer_package_prevention(
...
+
+        if layer_source.is_file and not layer_source.is_localized_path:
+            layer_path = Path(layer_source.layer.source()).resolve()
+
+            if home_path not in layer_path.parents:
+                return FeedbackResult(
+                    self.tr(
+                        "Layer '{}' is outside the project's home directory."
+                        "This layers may cause issues."
+                        "Please move the layer within '{}'."
```
Comment (suricactus) [inline libqfieldsync/project_checker.py:432]:
```suggestion
                        'Layer "{}" is outside the project's home directory.'
                        "This layers may cause issues."
                        'Please move the layer within "{}".'
```

I think pretty much all the code shows double quotes to the end users.

**Review by suricactus — APPROVED** [review]
Amazing work! Ready to merge when the small comment is fixed. Please squash to a single commit and force push.

File: libqfieldsync/project_checker.py (hunk 1 above)
Comment (SeqLaz) [reply libqfieldsync/project_checker.py]: Great, really thanks!

**Comment by SeqLaz** [comment]
> Amazing work! Ready to merge when the small comment is fixed. Please squash to a single commit and force push.

Hey @suricactus, I think that now is ready to be merge!

File: libqfieldsync/project_checker.py [hunk 2]
```diff
@@ -410,3 +415,22 @@ def check_layer_package_prevention(
...
 
         return None
+
+    def check_external_layers(
+        self, layer_source: LayerSource
+    ) -> Optional[FeedbackResult]:
+        home_path = Path(self.project.fileName()).parent
+
+        if layer_source.is_file and not layer_source.is_localized_path:
+            layer_path = Path(layer_source.layer.source()).resolve()
```
Comment (m-kuhn) [inline libqfieldsync/project_checker.py:425]: This seems to be causing some hiccups, see https://github.com/opengisch/qfieldsync/issues/630?notification_referrer_id=NT_kwDOAAj6d7IxMzc0MTMwNDc4Mzo1ODg0MDc#issuecomment-2525185190

The source is more than just a path

Comment (suricactus) [reply libqfieldsync/project_checker.py:425]: My bad I overlooked this during the review. Thanks for the pointer!

**Pull request merged**